packages/financial-templates-lib/test/price-feed/TraderMadePriceFeed.js

```diff
@@ -0,0 +1,303 @@
+
+    await traderMadePriceFeed.update();
+
+    // Should return the current price in the data.
+    assert.equal(traderMadePriceFeed.getCurrentPrice().toString(), toWei("0.1553"));
+  });
+
+  it("Last update time", async function() {
+    // Inject data.
+    networker.getJsonReturns = [...validResponses];
+
+    await traderMadePriceFeed.update();
+
+    // Should return the mock time.
+    assert.equal(traderMadePriceFeed.getLastUpdateTime(), mockTime);
+  });
+
+  it("No or bad response", async function() {
+    // Bad price response.
+    networker.getJsonReturns = [
+      {
+        quotes: [
+          {
+            error: "test"
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            close: 0.1543,
+            date: "2021-01-26 00:00:00"
+          },
+          {
+            close: 0.1533,
+            date: "2021-01-26 00:00:00"
+          },
+          {
+            close: 0.1523,
+            date: "2021-01-26 00:00:00"
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            close: 0.1543,
+            date: "2021-01-26 00:00:00"
+          },
+          {
+            close: 0.1533,
+            date: "2021-01-26 00:00:00"
+          },
+          {
+            close: 0.1523,
+            date: "2021-01-26 00:00:00"
+          }
+        ]
+      }
+    ];
+
+    // Update should throw errors in both cases.
+    assert.isTrue(await traderMadePriceFeed.update().catch(() => true), "Update didn't throw");
+
+    assert.equal(traderMadePriceFeed.getCurrentPrice(), undefined);
+    assert.equal(traderMadePriceFeed.getHistoricalPrice(1614319100), undefined);
+    assert.equal(traderMadePriceFeed.getHistoricalPricePeriords(), undefined);
+
+    // Bad historical ohlc response.
+    networker.getJsonReturns = [
+      {
+        quotes: [
+          {
+            ask: 0.1553
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            error: "test"
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            error: "test"
+          }
+        ]
+      }
+    ];
+
+    assert.isTrue(await traderMadePriceFeed.update().catch(() => true), "Update didn't throw");
+
+    assert.equal(traderMadePriceFeed.getCurrentPrice(), undefined);
+    assert.equal(traderMadePriceFeed.getHistoricalPrice(1614319100), undefined);
+    assert.equal(traderMadePriceFeed.getHistoricalPricePeriords(), undefined);
+
+    // Inverted price feed returns undefined for prices equal to 0 since it cannot divide by 0
+    networker.getJsonReturns = [
+      {
+        quotes: [
+          {
+            ask: 0
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            error: "test"
+          }
+        ]
+      },
+      {
+        quotes: [
+          {
+            error: "test"
+          }
+        ]
+      }
+    ];
+
+    assert.isTrue(await traderMadePriceFeed.update().catch(() => true), "Update didn't throw");
```

Comment:
You can also use `assert.throws()`

Reply:
`assert.throws` for `async` function may be a little complex, can we skip this?

Reply:
Yep

Reply:
@mrice32 btw this is how you can async test for throws

Reply:
The way you have it now looks right to me!